auth/src/main/java/com/firebase/ui/auth/ui/provider/EmailProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.firebase.ui.auth.ui.provider;
+
+import android.content.Intent;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.AuthUI;
+import com.firebase.ui.auth.ErrorCodes;
+import com.firebase.ui.auth.FirebaseUiException;
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.ui.HelperActivityBase;
+import com.firebase.ui.auth.ui.email.EmailActivity;
+import com.firebase.ui.auth.viewmodel.RequestCodes;
+import com.firebase.ui.auth.viewmodel.idp.ProvidersHandlerBase;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class EmailProvider extends ProviderBase {
+    public EmailProvider(ProvidersHandlerBase handler) {
+        super(handler);
+    }
+
+    @NonNull
+    @Override
+    public String getName() {
```

Comment:
Do we ever call `getName()` from somewhere that we don't have `Context`?  Would prefer to pass it in here rather than rely on `getApplicationContext()` more and more.  Maybe I'm being irrational though?

Other option would be to have this return `@StringRes int` rather than `String`.